File: backend-java/src/main/java/org/egualpam/contexts/observable/shared/application/domain/DomainEventId.java
package org.egualpam.contexts.observable.shared.application.domain;

import org.egualpam.contexts.observable.shared.application.domain.exceptions.InvalidDomainEntityId;

import java.util.UUID;

import static java.util.UUID.randomUUID;

public record DomainEventId(String value) {
    public DomainEventId {
        try {
            UUID.fromString(value);
        } catch (IllegalArgumentException e) {
            throw new InvalidDomainEntityId(value);
        }
    }

    public static DomainEventId generate() {
        return new DomainEventId(randomUUID().toString());
    }
}

File: backend-kotlin/src/main/kotlin/org/egualpam/contexts/observable/shared/application/domain/DomainEventId.kt
package org.egualpam.contexts.observable.shared.application.domain

import org.egualpam.contexts.observable.shared.application.domain.exceptions.InvalidDomainEntityId
import java.util.UUID

data class DomainEventId(val value: String) {
  init {
    try {
      UUID.fromString(value)
    } catch (e: IllegalArgumentException) {
      throw InvalidDomainEntityId(value)
    }
  }

  companion object {
    fun generate() = DomainEventId(UUID.randomUUID().toString())
  }
}
